portal-frontend/src/app/features/edit-application/application-owner-dialog/application-owner-dialog.component.ts

```diff
@@ -0,0 +1,116 @@
+import { MatButtonToggleChange } from '@angular/material/button-toggle';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import {
+  ApplicationOwnerCreateDto,
+  ApplicationOwnerDto,
+  ApplicationOwnerUpdateDto,
+} from '../../../services/application-owner/application-owner.dto';
+import { ApplicationOwnerService } from '../../../services/application-owner/application-owner.service';
+
+@Component({
+  selector: 'app-application-owner-dialog',
+  templateUrl: './application-owner-dialog.component.html',
+  styleUrls: ['./application-owner-dialog.component.scss'],
+})
+export class ApplicationOwnerDialogComponent {
+  type = new FormControl<string | null>('INDV');
+  firstName = new FormControl<string | null>('', [Validators.required]);
+  lastName = new FormControl<string | null>('', [Validators.required]);
+  organizationName = new FormControl<string | null>('');
+  phoneNumber = new FormControl<string | null>('', [Validators.required]);
+  email = new FormControl<string | null>('', [Validators.required, Validators.email]);
+
+  isEdit = false;
+  existingUuid: string | undefined;
+
+  form = new FormGroup({
+    type: this.type,
+    firstName: this.firstName,
+    lastName: this.lastName,
+    organizationName: this.organizationName,
+    phoneNumber: this.phoneNumber,
+    email: this.email,
+  });
+
+  constructor(
+    private dialogRef: MatDialogRef<ApplicationOwnerDialogComponent>,
+    private appOwnerService: ApplicationOwnerService,
+    @Inject(MAT_DIALOG_DATA)
+    public data: {
+      fileId?: string;
+      parcelUuid?: string;
+      existingOwner?: ApplicationOwnerDto;
+    }
+  ) {
+    if (data && data.existingOwner) {
+      this.onChangeType({
+        value: data.existingOwner.type.code,
+      } as any);
+      this.isEdit = true;
+      this.type.setValue(data.existingOwner.type.code);
+      this.firstName.setValue(data.existingOwner.firstName);
+      this.lastName.setValue(data.existingOwner.lastName);
+      this.organizationName.setValue(data.existingOwner.organizationName);
+      this.phoneNumber.setValue(data.existingOwner.phoneNumber);
+      this.email.setValue(data.existingOwner.email);
+      this.existingUuid = data.existingOwner.uuid;
+    }
+  }
+
+  onChangeType($event: MatButtonToggleChange) {
+    if ($event.value === 'ORGZ') {
+      this.organizationName.setValidators([Validators.required]);
+      this.firstName.setValidators([]);
+      this.lastName.setValidators([]);
+      this.firstName.reset();
+      this.lastName.reset();
+    } else {
+      this.organizationName.setValidators([]);
+      this.firstName.setValidators([Validators.required]);
+      this.lastName.setValidators([Validators.required]);
+      this.organizationName.reset();
+    }
+  }
+
+  async onCreate() {
+    if (!this.data.fileId) {
+      console.error('ApplicationOwnerDialogComponent misconfigured, needs fileId for create');
```

Comment:
I wonder if this should show an error message to user so they cold try reload the page or something

Reply:
If a user sees this error it would be a mistake on our part and it would require a code fix.

Reply:
The thing is that the user will not see this one since it is the console. But I guess this error should never occur since this page can only be retrieved from application